src/app/login/login.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import {Router} from '@angular/router';
+import { LoginResponse } from '../api.model';
+import { AuthService } from '../auth.service';
+
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+
+  public email: string;
+  public password: string;
+  public error = false;
+
+  constructor(
+    public readonly authService: AuthService,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    if (this.authService.authTokenIsPresent()) {
+      if (this.authService.authTokenHasExpired()) {
+        this.refreshToken();
+      } else {
+        this.redirectToHome();
+      }
+    }
```

Comment:
E' la seconda volta che vedo questo, penso che... dovremmo forse encapsularlo in una funzione e riusarla anziche' creare ridondanza di codice